walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,96 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
 
+/**
+ * Implements the "find a meeting" feature which returns the possible times when
+ * a meeting could be scheduled given information about the meeting and a list
+ * of all the events already scheduled.
+ */
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<String> attendees = request.getAttendees();
+
+    // There are no options for a meeting longer than a day.
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+      return Arrays.asList();
+    }
+
+    // Get the list of time ranges where the attendees have scheduled events.
+    List<TimeRange> conflicts = findTimeConflicts(events, attendees);
+
+    return getPossibleMeetingTimes(conflicts, request);
+  }
+
+  /**
+   * Extracts a list of time ranges where any of the given people have events
+   * scheduled. Returns the list sorted by start time in ascending order.
+   */
+  private List<TimeRange> findTimeConflicts(Collection<Event> events, Collection<String> people) {
```

Comment:
For method return types, you should generally use the immutable type (such as ImmutableList) instead of the general collection interface type (such as List). This communicates to your callers the semantic guarantees of immutable types, which is almost always very useful information. When it's a general collection type, callers may be under the impression that they can mutate the return value, leading to failures at runtime.